rust/connlib/tunnel/src/utils.rs

```diff
@@ -85,3 +85,55 @@ pub(crate) fn ipv6(ip: IpNetwork) -> Option<Ipv6Network> {
         IpNetwork::V6(v6) => Some(v6),
     }
 }
+
+/// Helper container to aggregate candidates and emit them in sorted order, starting with the highest-priority one (host candidates).
+///
+/// Whilst no fatal in theory, emitting candidates in the wrong order can cause temporary connectivity problems.
+/// `str0m` needs to "replace" candidates when it receives better ones which can cancel in-flight STUN requests.
+/// By sorting the candidates by priority, we try the best ones first, being symphatic to the ICE algorithm.
+#[derive(Default)]
+pub(crate) struct Candidates(Vec<snownet::Candidate>);
+
+impl Candidates {
+    pub(crate) fn push(&mut self, candidate: snownet::Candidate) {
+        self.0.push(candidate)
+    }
+
+    pub(crate) fn serialize(mut self) -> BTreeSet<String> {
+        self.0.sort_by(priority_desc);
+
+        self.0.into_iter().map(|c| c.to_sdp_string()).collect()
+    }
```

Comment:
can we keep it sorted on `push` then we don't need a custom `serialize`?

Reply:
You mean implementing `Display` and using `to_string`?

Reply:
I was thinking more about `#[derive(Serialize)]` with a `#[serde(serialize_with = "to_sdp_string")]`.

It would  need a bit of work since we need the `to_sdp_string` to work over the whole `Vec` but I think it wouldn't be hard? though I might be wrong

Reply:
And then you'd use that collection directly in the portal message? That is a good idea. I'll have a play.

Reply:
I can't do it right now because I don't want to depend on `snownet` in `connlib_shared` where the message sit.